readthedocs/projects/tasks/mixins.py

```diff
@@ -45,10 +45,12 @@ def sync_versions(self, vcs_repository):
         # Do not use ``ls-remote`` if the VCS does not support it or if we
         # have already cloned the repository locally. The latter happens
         # when triggering a normal build.
-        use_lsremote = (
-            vcs_repository.supports_lsremote
-            and not vcs_repository.repo_exists()
-        )
+        # ALWAYS use lsremote when we have GIT_CLONE_FETCH_CHECKOUT_PATTERN on.
+        # The new pattern does not fetch branch and tag data, so we always
+        # have to do ls-remote.
+        use_lsremote = self.data.project.has_feature(
+            Feature.GIT_CLONE_FETCH_CHECKOUT_PATTERN
+        ) or (vcs_repository.supports_lsremote and not vcs_repository.repo_exists())
```

Comment:
This is related, yes.

The old code has the following assumption: When performing `git clone` or `git fetch`, we are using `--no-single-branch` and `--tags` respectively. So we can use the previons `.branches` and `.tags` methods that will read branch and tag data from the cloned repo.

Since we are no longer doing this with the new pattern, we need to always rely on `ls-remote`.

LMK if I missed something.

Reply:
In that case, we should check if the repo supports ls-remote too, otherwise we may break other VCS repos (I know we are deprecating them, but we have a timeline for it). This is only relevant if we want to test this feature flag more broadly without selecting individual projects of course.

```python
use_clone_fetch_for_checkout = has_feature and vcs_repository.supports_lsremote
```

Reply:
Thanks @stsewd :+1: 

Fixed in f1d426b36

I think we should test the feature narrowly on specific projects, then more broadly. So it will overlap with other VCS support :+1: Even if we are deprecating other VCS support, I understand that we'll still keep the code around for a little while "just in case".